spring-cloud-pivotal-core/src/main/java/org/springframework/cloud/pivotal/service/common/GemfireServiceInfo.java

```diff
@@ -0,0 +1,65 @@
+package org.springframework.cloud.pivotal.service.common;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.springframework.cloud.service.BaseServiceInfo;
+import org.springframework.cloud.service.ServiceInfo.ServiceLabel;
+
+/**
+ * 
+ * Information to access Gemfire services
+ * 
+ * @author Vinicius Carvalho
+ *
+ */
+@ServiceLabel("gemfire")
+public class GemfireServiceInfo extends BaseServiceInfo {
+	final Pattern p = Pattern.compile("(.*)\\[(\\d*)\\]");
+	private URI[] locators;
+	private Map<String, Object> credentials;
+
+	public GemfireServiceInfo(String id, Map<String, Object> credentials) {
+		super(id);
+		this.credentials = credentials;
+		List<String> locatorList = (List<String>) credentials.get("locators");
+		this.locators = new URI[locatorList.size()];
+		for (int i = 0; i < locatorList.size(); i++) {
+			locators[i] = parseLocators(locatorList.get(i));
+		}
+	}
+
+	URI parseLocators(String locator) throws IllegalArgumentException {
```

Comment:
I've asked myself that question. It's a weird syntax IMHO, but its how GF uses it (even at GFSH it's how we connect to a locator: connect locator=host[port]. Maybe a legacy thing? I like the URI syntax better for cups, I don't see why not support both, this can be added for sure